### PR TITLE
feat: adding a autosync_enabled field  to the argocd_app_info gauge

### PR DESCRIPTION
**Pull request by gerardocorea**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Closes [ISSUE #12565]

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: controller/metrics/metrics.go [hunk 1]
```diff
@@ -381,7 +381,13 @@ func (c *appCollector) collectApps(ch chan<- prometheus.Metric, app *argoappv1.A
 		healthStatus = health.HealthStatusUnknown
 	}
 
-	addGauge(descAppInfo, 1, git.NormalizeGitURL(app.Spec.GetSource().RepoURL), app.Spec.Destination.Server, app.Spec.Destination.Namespace, string(syncStatus), string(healthStatus), operation)
+	autoSyncEnabled := false
+
+	if app.Spec.SyncPolicy != nil {
+		autoSyncEnabled = app.Spec.SyncPolicy.Automated.SelfHeal
```
Comment (jannfis) [inline controller/metrics/metrics.go:387]: Both, `app.Spec.SyncPolicy` and `app.spec.SyncPolicy.Automated` are pointers and can be `nil`. Also, SelfHeal is not the correct value.
```suggestion
		autoSyncEnabled = app.Spec.SyncPolicy != nil && app.Spec.SyncPolicy.Automated != nil
```

**Review by jannfis — CHANGES_REQUESTED** [review]
Thanks for your PR! 

There's some concern, PTAL at my comment

File: controller/metrics/metrics.go (hunk 1 above)
Comment (gerardocorea) [reply controller/metrics/metrics.go]: Updated pr to reflect recommended changes, thank you!

File: controller/metrics/metrics.go [hunk 2]
```diff
@@ -381,7 +381,11 @@ func (c *appCollector) collectApps(ch chan<- prometheus.Metric, app *argoappv1.A
 		healthStatus = health.HealthStatusUnknown
 	}
 
-	addGauge(descAppInfo, 1, git.NormalizeGitURL(app.Spec.GetSource().RepoURL), app.Spec.Destination.Server, app.Spec.Destination.Namespace, string(syncStatus), string(healthStatus), operation)
+	autoSyncEnabled := false
+
+	autoSyncEnabled = app.Spec.SyncPolicy != nil && app.Spec.SyncPolicy.Automated != nil
```
Comment (ishitasequeira) [inline controller/metrics/metrics.go:386]: nit: we can replace this 2 lines with below:

```suggestion

	autoSyncEnabled := app.Spec.SyncPolicy != nil && app.Spec.SyncPolicy.Automated != nil
```

Comment (gerardocorea) [reply controller/metrics/metrics.go]: Appreciate the feedback! Pushed a commit

**Review by ishitasequeira — APPROVED** [review]

**Review by jannfis — APPROVED** [review]
LGTM, thank you!

**Pull request merged**